src/go/rdctl/pkg/factoryreset/factory_reset_windows.go

```diff
@@ -30,10 +32,56 @@ import (
 	"golang.org/x/text/encoding/unicode"
 )
 
-func KillRancherDesktop() {
+func KillRancherDesktop() error {
 	cmd := exec.Command("taskkill", "/IM", "Rancher Desktop.exe", "/T", "/F")
 	cmd.SysProcAttr = &syscall.SysProcAttr{CreationFlags: CREATE_NO_WINDOW}
-	cmd.Run()
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("Failed to taskkill Rancher Desktop.exe: %w", err)
+	}
+	val := os.Getenv("RANCHER_DESKTOP_CHILD")
+	if val != "1" {
+		return nil
+	}
+	// Now verify that the taskkill really killed it.
```

Comment:
So this _only_ runs as a child of the GUI. Which means any messages only go to the logs… is that correct? I'm having trouble deciding what the intended behaviour is around the environment variable.